pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -261,6 +265,47 @@ func (r *Reconciler) create() error {
 		Items: metadataItems,
 	}
 
+	// check to see if this is a windows instance
+	// if yes, we need to get the powershell boot script from the provider spec
+	// or the `windows-user-data` secret in the `openshfit-machine-api` namespace.
+	// see https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/secrets/secrets.go
+	osid, found := r.machine.Labels["machine.openshift.io/os-id"]
+	if found && osid == "Windows" {
+		script := r.getWindowsScriptFromProviderSpec()
+
+		if len(script) == 0 {
+			// didn't find in provider spec, look for secret
+			script, err = r.getWindowsScriptFromSecret()
+			if err != nil {
+				return fmt.Errorf("error getting Windows script secret data: %v", err)
+			}
+		}
+
+		if len(script) > 0 {
+			// found a script, we need to update the instance data that will be sent to GCP
+			found := false
+			for i, item := range instance.Metadata.Items {
+				if item.Key == windowsScriptMetadataKey {
+					instance.Metadata.Items[i].Value = &script
+					found = true
+					break
+				}
+			}
+
+			if !found {
+				// if we did not find this key in the previous metadata, add it now
+				instance.Metadata.Items = append(instance.Metadata.Items, &compute.MetadataItems{
+					Key:   windowsScriptMetadataKey,
+					Value: &script,
+				})
+			}
+		} else {
+			// we didn't find a script
+			return machinecontroller.CreateMachine("cannot find Windows script in providerSpec or Secret")
```

Comment:
For my own education; does `machinecontroller.CreateMachine()` throws an exception/error?

Reply:
@jrvaldes that function does wrap the creation of an `Error` type. it has an unfortunate name =(